salt/config/__init__.py

```diff
@@ -3649,6 +3653,21 @@ def get_id(opts, cache_minion_id=False):
     if opts.get('minion_id_lowercase'):
         newid = newid.lower()
         log.debug('Changed minion id %s to lowercase.', newid)
+
+    # Optionally remove a domain or many in a generated minion id
+    rem_domain = opts.get('minion_id_remove_domain')
+    if rem_domain:
+        if rem_domain is True:
+            if '.' in newid:
+                # Clean away any domain
+                newid, xdomain = newid.split('.', 1)
+                log.debug('Cleaned away any domain (%s) from minion id.', xdomain)
```

Comment:
"Cleaned away" doesn't make a lot of sense here, you can use "Removed" as in the log message you added below.

Reply:
Hi @terminalmage, I implemented the changes you requested (test cases for True and False, closed PR 49213 and changed the wording). 

Could you please comment or approve?